### PR TITLE
Add support to turn off az/za domain checking

### LINKED ISSUE #388: Add handling of check_azza_domain parameter for UVBeams
To allow for faster interpolation. See https://github.com/RadioAstronomySoftwareGroup/pyuvdata/pull/1144

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

This adds a parameter to `run_uvsim` and `run_uvdata_uvsim` to allow for the check that the beam covers the interpolation location to be turned off. The default behavior is to turn off the check if the beam covers the full sky horizon to horizon.

Turning off this check can lead to substantial speedups, but if sources are simulated outside the beam area the response will be incorrect.

This requires updating out pyuvdata requirement to >= 2.2.8 when the `check_azza_domain` parameter was added to `UVBeam.interp`. I don't see a downside to updating this requirement.

## Motivation and Context

closes #388 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Reference simulation update or replacement
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

For all pull requests:
- [x] I have read the [contribution guide](CONTRIBUTING.md).
- [x] My code follows the code style of this project.

New feature checklist:
- [x] I have added or updated the docstrings associated with my feature using the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] I have updated the documentation to highlight my new feature (if appropriate).
- [x] I have added tests to cover my new feature.
- [x] All new and existing tests pass.
- [ ] I have checked that I reproduce the reference simulations or if there are differences they are explained below (if appropriate). If there are changes that are correct, I will update the reference simulation files after this PR is merged.
- [ ] I have checked (e.g., using the benchmarking tools) that this change does not significantly increase typical runtimes. If it does, I have included a justification in the comments on this PR.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvsim/blob/master/CHANGELOG.md).

### CODE REVIEW

File: pyuvsim/analyticbeam.py [hunk 1]
```diff
@@ -115,7 +115,15 @@ def efield_to_power(self):
...
 
-    def interp(self, az_array, za_array, freq_array, reuse_spline=None, spline_opts=None):
+    def interp(
+        self,
+        az_array,
+        za_array,
+        freq_array,
+        reuse_spline=None,
+        spline_opts=None,
+        check_azza_domain=None
```
Comment (steven-murray) [inline pyuvsim/analyticbeam.py:125]: Should we not just use `**kwargs` in the signature and then just ignore them? I imagine there might be more additions to `interp()` in pyuvbeam over time, and we don't always want to have to remember to add the parameter to the analytic beam.

File: pyuvsim/utils.py [hunk 2]
```diff
@@ -270,24 +267,19 @@ def write_uvdata(
     if noclobber:
         outfile_name = check_file_exists_and_increment(outfile_name)
 
-    if version.parse(pyuvdata.__version__) >= version.parse("2.2.7"):
-        write_kwargs = {'fix_autos': fix_autos}
-    else:
-        write_kwargs = {}
-
     print('Outfile path: ', outfile_name, flush=True)
```
Comment (steven-murray) [inline pyuvsim/utils.py:270]: I know this isn't a part of this PR, but I don't like bare print statements. Should this be a logging statement? At least then the user can control its output.

**Review by steven-murray — COMMENTED** [review]
Thanks @bhazelton, this is well done. I have only a couple of small comments. Looks like coverage needs improvement as well.

File: pyuvsim/uvsim.py [hunk 3]
```diff
@@ -572,11 +590,30 @@ def _check_ntasks_valid(Ntasks_tot):
         )
 
 
+def _check_all_azza_beams_full_sky(beam_list):
```
[the remark this answers is not in the document]
Comment (bhazelton) [reply pyuvsim/uvsim.py:593]: Sure, I didn't think about it being more widely useful. I was also debating moving it to the BeamList object.

File: pyuvsim/utils.py (hunk 2 above)
Comment (bhazelton) [reply pyuvsim/utils.py:270]: probably, I'll take a look at it.

File: pyuvsim/analyticbeam.py (hunk 1 above)
Comment (bhazelton) [reply pyuvsim/analyticbeam.py:125]: sure, that's probably reasonable. Longer term this should move to pyuvdata and we can think better about how to handle these there.

File: pyuvsim/uvsim.py [hunk 4]
```diff
@@ -572,11 +590,30 @@ def _check_ntasks_valid(Ntasks_tot):
...
+    for beam in beam_list:
+        if isinstance(beam, UVBeam) and beam.pixel_coordinate_system == "az_za":
+            # regular gridding is enforced in UVBeam checking, so can use first diff
+            axis1_diff = np.diff(beam.axis1_array)[0]
+            axis2_diff = np.diff(beam.axis2_array)[0]
+            max_axis_diff = np.max([axis1_diff, axis2_diff])
+            if not (
+                np.max(beam.axis2_array) >= np.pi / 2. - max_axis_diff * 2.0
+                and np.min(beam.axis1_array) <= max_axis_diff * 2.0
+                and np.max(beam.axis1_array) >= 2. * np.pi - max_axis_diff * 2.0
```
[the remark this answers is not in the document]
Comment (bhazelton) [reply pyuvsim/uvsim.py:604]: Actually, the regular grid requirement in the UVBeam check may ensure that there are no holes.

**Comment by bhazelton** [comment]
@steven-murray I think I addressed your comments, but now the hera_sim tests are failing and I don't understand why. The error message is
```
TypeError: interp() got an unexpected keyword argument 'check_azza_domain'
```
Which should not be happening because the pyuvdata version is 2.2.8, which includes the change.

**Review by steven-murray — APPROVED** [review]

**Pull request merged**